pkg/acpi/tables_linux.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build linux
+
+package acpi
+
+import (
+	"path/filepath"
+)
+
+var (
+	Methods = map[string]TableMethod{
+		"files": RawTablesFromSys,
+	}
+)
+
+// MethodNames returns the list of supported MethodNames.
+func MethodNames() string {
```

Comment:
i'm a little lost in this methods thing -- what's this for?

Reply:
sysfs is not always guarantted to be there or even useful. I am probably going to have to add EBDA and RSDP methods. On OSX and other systems, there will be other methods. The string to Method map makes it easier across kernels and configurations to specify different ways of extracting tables.

Reply:
oh, sorry, i should have been more clear.

why do we have this naming thing though? can't users of the library just call the functions for each method?

if the command needs the names, i'd move it to the cmd

Reply:
Yeah I can move it to the command, but does it make sense to have things like this:
acpicat_linux.go
//build +linux
var methods map {
name
func}{
"ebda": acpi.EBDA,
"files: acpi.Files,
}

acpicat_darwin.go
var methods map {...}{"sys": acpi.Sys}

acpicat_freebsd.go
etc.

All this in acpicat *command*? doesn't this belong in the package? 

and so on. I don't think it does because it does not insulate the command from changes in the package. The knowledge of the types of operations and how to implement them ought to be in the package I believe.

Reply:
I think for that, there should be a thing like the GetRSDP method that cycles through them. Otherwise, if they don't want to use em all, if they have a specific one in mind, the user still has to know which strings are in the map and what they mean, at which point I'd rather they know the actual function names

Reply:
I can't agree. This is structured in the way I would want to use these tools.

But I'll take this off the board, I'm probably the only user.

Reply:
We should at least be consistent. Since there is a GetRSDP that cycles, there should be one for GetTables, too.

Reply:
Look we can agree to disagree at some point and commit it, but I'm still curious about why you disagree, and I don't think you should just give up.

Reply:
will add GetTables, that's an excellent point.